### PR TITLE
Custom rules / goodcheck

### PR DESCRIPTION
**Pull request by ivanha09**
This section is to help users with custom rules and goodcheck by providing examples of rules and basic instructions.

### CODE REVIEW

**Comment by ybiquitous** [comment]
Here are my opinions:

- How about adding links to each page with <https://help.sider.review/tools/others/goodcheck>?
- `Custom Tools` is confusing with `Analysis Tools` to me, so how about renaming to `Custom Rules`?

image

File: docs/custom-tools/goodcheck.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+* `justification`: a sequence of strings to tell writers of exceptions to the rule.
+* `glob`: a *glob* or a sequence of *glob*s.
+* `pass`: a string, or a sequence of strings. A test that should not match the given pattern.
+* `fail`: a string, or a sequence of strings. A test that should match the given pattern.
+
+### Rule: "_blank" Security Issue
+When `target = "_blank"` is used, the opened page can access the original window object and potentially redirect the original page to a malicious URL. In this example, the rule will look for patterns of `"_blank"` and suggest to use `rel="noopener"` to prevent the opened page from having access.
+
+```yaml
+ - id: security.link
```
Comment (ybiquitous) [inline docs/custom-tools/goodcheck.md]: [IMO] How about adding `rules:` to the code snippet? If users copy and paste the snippet, I think it's useful that the snippet will work.

```yaml
rules:
  - id: security.link
    # ...

Comment (ivanha09) [reply docs/custom-tools/goodcheck.md]: Yeah. 'Custom Rules' makes more sense I'll change it to that.

Comment (ivanha09) [reply docs/custom-tools/goodcheck.md]: I think adding the 'rules:' will be helpful. 

I'll remove the sample yml file section on this page then. 

Is there a good way of explaining that the 'rules:' is only required at the top of the file? I was thinking of putting it in **Basic instructions**. 3. Write your own rule or copy rule(s) into the goodcheck.yml file.('rules:' is only needed as the first line of the 'goodcheck.yml' file.)

File: website/sidebars.json [hunk 2]
```diff
@@ -88,6 +88,9 @@
         ]
       }
     ],
+    "Custom Tools": [
```
Comment (soutaro) [inline website/sidebars.json]: I'd like to name the section `Custom rules`. We are going to explain the feature of Sider, not about the tools.

I also plan to add a `Tips & tricks` document in this section, which explains:

- The best practices to find rules.
- How to share the rules among repositories.

**Comment by soutaro** [comment]
@ivanha09 Click the `Merge pull request` button. 🎉

**Pull request merged**